solr/solrj/src/java/org/apache/solr/cluster/api/Config.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.cluster.api;
+
+import java.io.InputStream;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import org.apache.solr.common.SolrException;
+
+public interface Config {
```

Comment:
Use `ConfigSet` instead? `Config` is soooo overloaded.

Reply:
I don't like both names , `Config` & `ConfigSet`

`configset` suggests it is a set of configs? in reality it's a single configuration which has multiple files